### DISCUSSION TITLE
system attribute missing in derivation

### DISCUSSION THREAD

**Discussion by chrichrichri** [nix-community/dream2nix#1018] [Q&A] [answered]
Hello,

I am trying to install a package I created using dream2nix (from a python package) inside a docker container running nix. The docker image is derived from [nix docker image](https://github.com/NixOS/nix/blob/master/docker.nix) using a flake similar to this one [found here](https://icewind.nl/entry/gitea-actions-nix/). Everything works fine, until I try to install the package generated by dream2nix:

```
extraPkgs = with pkgs; [
  <input>.packages.${pkgs.system}.<my_package_name>
];

```

This error shows up. It seems in the `derivation` the `system` attribute is missing, but I was not able to find how to add it. Can anyone give a hint? Thanks!

```
[ERROR]    error: attribute 'system' missing
[ERROR]   
[ERROR]   at /nix/store/pk6hi49dmdbspimwxr64abvc0frz2wjr-source/docker.nix:180:25:
[ERROR]   
[ERROR]   179|             outPath = "${drv}";
[ERROR]   180|             system = "${drv.system}";
[ERROR]   |                         ^
[ERROR]   181|             type = "derivation";
[ERROR]  Child process exited with error code: 1
```

**Comment by DavHau** [comment, marked as answer]
You could set `public.system` in your dream2nix module. Mayeb we should have this by default. Feel free to open a PR

**Reply by chrichrichri** [reply]
Thank you that helped. I would be glad to open a PR - but honestly I am not sure if I would actually help with it since I am not so good at nix... :(
